shortcodes/epfl_news/utils.php

```diff
@@ -0,0 +1,96 @@
+<?php 
+
+/**
+ * Get url channel
+ * 
+ * $data: all data from back-end
+ */
+function epfl_news_get_url_channel($data) {
+    $url_channel = "";
+    if (count($data) > 0) {
+        $channel = $data[0]->channel->name;
+        $url_channel = "https://actu.epfl.ch/search/" . $channel;
+    }
+    return $url_channel;
+}
+
+/**
+ * Get image description
+ * 
+ * $news: news to display
+ */
+function epfl_news_get_image_description($news) {
+    if (get_locale() == 'fr_FR') {
+        $image_description = $news->fr_description;
+    } else {
+        $image_description = $news->en_description;
+    }
+    return $image_description;
+}
+
+/**
+ * Get label category
+ * 
+ * $news: news to display
+ */
+function epfl_news_get_label_category($news) {
+    if (get_locale() == 'fr_FR') {
+        $label_category = $news->category->fr_label;
+    } else {
+        $label_category = $news->category->en_label;
+    }
+    return $label_category;
+}
+
+/**
+ * Get publish date
+ * 
+ * $news: news to display
+ */
+function epfl_news_get_publish_date($news) {
+    $publish_date = new DateTime($news->publish_date);
+    $publish_date = $publish_date->format('d.m.y');
+    return $publish_date;
+}
+
+/**
+ * Get subtitle
+ * 
+ * $news: news to display
+ */
+function epfl_news_get_subtitle($news) {
+    $subtitle = strip_tags($news->subtitle);
+    return $subtitle;
+}
+
+/**
+ * Get visual url
+ * 
+ * $news: news to display
+ */
+function epfl_news_get_visual_url($news) {
+    $visual_url = substr($news->visual_url, 0, -11) . '1296x728.jpg';
+    return $visual_url;
+}
+
+/**
+ * Get media url
+ * 
+ * $news: news to display
+ */
+function epfl_news_get_media_url($news) {
+    $slug = str_replace("https://actu.epfl.ch/news/", "", $news->news_url);
+    $video_name = "teaser_" . $slug;
+          
+    global $wpdb;
```

Comment:
Suis moyennement à l'aise avec l'utilisation de requêtes explicites dans la DB.
J'ai trouvé ceci qui pourrait faire l'affaire, je te laisse décider : https://stackoverflow.com/questions/20470853/get-the-url-file-by-name-of-image

Si tu décides de prendre ça, tu peux même potentiellement l'ajouter à `utils.php`

Reply:
J'ai suivi ta proposition. Je n'utilise plus de requête explicite dans la DB.